AmplifyPlugins/DataStore/AWSDataStoreCategoryPlugin/AWSDataStorePlugin+DataStoreBaseBehavior.swift

```diff
@@ -82,13 +84,15 @@ extension AWSDataStorePlugin: DataStoreBaseBehavior {
     public func delete<M: Model>(_ modelType: M.Type,
                                  withId id: String,
                                  completion: @escaping DataStoreCallback<Void>) {
+        reinitStorageEngineIfNeeded()
```

Comment:
Instead of doing it lazy-init-style here, can you do it eagerly, right before your `completion(.successfulVoid)` calls inside of `clear(...)`?

Reply:
Hmm! I would have liked to do what you suggested.. hmmm but... 

Based on the implementation of reinitStorageEngineIfNeeded(), I don't think this is what we want, because then we would be calling `storage.startSync()` which would attempt to restart the entire SyncEngine (so... then we'd potentially rehydrate the data base we just deleted)

If reinitStorageEngine were to be changed to only resolveTheStorageEngine (and not start the sync engine), we'd still have to call to startSyncEngine on any query/delete/subscription.  At this point, i'm not seeing a huge benefit one way or the other.